src/txn/txn_rollback_to_stable.c

```diff
@@ -354,7 +354,16 @@ __rollback_abort_row_ondisk_kv(
           __wt_timestamp_to_string(vpack->durable_start_ts, ts_string[0]),
           __wt_timestamp_to_string(vpack->start_ts, ts_string[1]),
           __wt_timestamp_to_string(rollback_timestamp, ts_string[2]));
-        return (__rollback_row_ondisk_fixup_key(session, page, rip, rollback_timestamp, true));
+        if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+            return (__rollback_row_ondisk_fixup_key(session, page, rip, rollback_timestamp, true));
+        else {
+            /*
+             * In-memory database don't have a history store to provide a stable update, so remove
```

Comment:
In case of in-memory don't we have the previous updates in the update chain? 
For eg:- if the update chain is like U6(60)->U5(50)->U3(30)->U2(20)->U1(10) and ondisk has U4(40).
Format I used to represent update is Un(timestamp).
In the above case if we have to rollback to stable with stable timestamp at 30. 
In this case when we remove U4, we cannot remove the key itself.

Reply:
Yes, all the previous updates are present in the update chain until they are obsolete. Scenarios where none of the updates in the update list satisfies the stable timestamp then the written on-disk version key should be removed. The added code is for those scenarios to check whether the on-disk version is stable according to the stable timestamp or not and then remove it.